spring-integration-amqp/src/main/java/org/springframework/integration/amqp/inbound/AmqpInboundGateway.java

```diff
@@ -103,7 +104,7 @@ public void onMessage(Message message, Channel channel) {
 				final org.springframework.messaging.Message<?> reply = sendAndReceiveMessage(request);
 				if (reply != null) {
 					// TODO: fallback to a reply address property of this gateway
-					Address replyTo = message.getMessageProperties().getReplyToAddress();
+					Address replyTo = AddressUtils.decodeReplyToAddress(message);
```

Comment:
M-m-m. Forgot to mantion: it works only with AMQP  >= 1.4.1.
Will hope that end-users follow with our transitive dependencies...